Classes/StenoSignal.sc

```diff
@@ -43,7 +51,22 @@ StenoSignal {
 
 	// get filter input
 	filterInput { |argNumChannels, offset = 0|
-		^this.quelleInput(argNumChannels, offset) * env
+		var sig, bus;
+		// if this synth is a replacement for another one, 
+		// read from alternative bus during attack time, switch to inBus afterwards
+		bus = Select.kr(replacement * (1-ToggleFF.kr(env < 1)), [inBus, fadeBus]);
+		sig = In.ar(bus, numChannels).asArray.drop(offset);
+
+		if(argNumChannels.notNil) {
+			sig = sig.keep(argNumChannels);
+			if(multiChannelExpand) { sig = sig.wrapExtend(argNumChannels) };
+		};
+
+		// if gate is released, write unprocessed input to fadeBus
+		ReplaceOut.ar(fadeBus, Select.ar(ToggleFF.kr(gate), [input, Silent.ar()]));
```

Comment:
I'm a little wary of `ToggleFF`. Why do we need it? Shouldn't it be `gate < 1` ?